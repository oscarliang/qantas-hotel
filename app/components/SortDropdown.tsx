import React from 'react';

type SortOption = 'high-low' | 'low-high';

type SortDropdownProps = {
    currentSort: SortOption;
    onSortChange: (sort: SortOption) => void;
};

const SortDropdown: React.FC<SortDropdownProps> = ({ currentSort, onSortChange }) => {
    return (
        <div className="flex items-center">
            <span className="text-sm font-bold mr-2">Sort by</span>
            <div className="relative">
                <select
                    value={currentSort}
                    onChange={(e) => onSortChange(e.target.value as SortOption)}
                    className="appearance-none bg-white border border-gray-300 text-gray-700 py-1 px-3 pr-8 rounded leading-tight focus:outline-none focus:border-blue-500"
                >
                    <option value="high-low">Price high-low</option>
                    <option value="low-high">Price low-high</option>
                </select>
                <div className="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700">
                    <svg className="fill-current h-4 w-4" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20">
                        <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" />
                    </svg>
                </div>
            </div>
        </div>
    );
};

export default SortDropdown; 