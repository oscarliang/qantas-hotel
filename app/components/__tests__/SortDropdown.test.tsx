import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import SortDropdown from '../SortDropdown';

describe('SortDropdown Component', () => {
    const mockOnSortChange = jest.fn();

    beforeEach(() => {
        mockOnSortChange.mockClear();
    });

    it('renders correctly with default props', () => {
        const { container } = render(
            <SortDropdown currentSort="high-low" onSortChange={mockOnSortChange} />
        );

        expect(screen.getByText('Sort by')).toBeInTheDocument();
        expect(screen.getByRole('combobox')).toHaveValue('high-low');
        expect(container).toMatchSnapshot();
    });

    it('shows both sorting options', () => {
        render(
            <SortDropdown currentSort="high-low" onSortChange={mockOnSortChange} />
        );

        const select = screen.getByRole('combobox');
        expect(select).toBeInTheDocument();

        // Check if options exist
        const options = screen.getAllByRole('option');
        expect(options).toHaveLength(2);
        expect(options[0]).toHaveValue('high-low');
        expect(options[1]).toHaveValue('low-high');
    });

    it('calls onSortChange when selection changes', () => {
        render(
            <SortDropdown currentSort="high-low" onSortChange={mockOnSortChange} />
        );

        const select = screen.getByRole('combobox');

        // Change selection to low-high
        fireEvent.change(select, { target: { value: 'low-high' } });

        expect(mockOnSortChange).toHaveBeenCalledTimes(1);
        expect(mockOnSortChange).toHaveBeenCalledWith('low-high');
    });

    it('renders with low-high selected correctly', () => {
        const { container } = render(
            <SortDropdown currentSort="low-high" onSortChange={mockOnSortChange} />
        );

        expect(screen.getByRole('combobox')).toHaveValue('low-high');
        expect(container).toMatchSnapshot();
    });
}); 