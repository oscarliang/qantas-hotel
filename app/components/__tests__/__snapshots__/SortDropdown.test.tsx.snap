// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortDropdown Component renders correctly with default props 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <span
      class="text-sm font-bold mr-2"
    >
      Sort by
    </span>
    <div
      class="relative"
    >
      <select
        class="appearance-none bg-white border border-gray-300 text-gray-700 py-1 px-3 pr-8 rounded leading-tight focus:outline-none focus:border-blue-500"
      >
        <option
          value="high-low"
        >
          Price high-low
        </option>
        <option
          value="low-high"
        >
          Price low-high
        </option>
      </select>
      <div
        class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
      >
        <svg
          class="fill-current h-4 w-4"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`SortDropdown Component renders with low-high selected correctly 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <span
      class="text-sm font-bold mr-2"
    >
      Sort by
    </span>
    <div
      class="relative"
    >
      <select
        class="appearance-none bg-white border border-gray-300 text-gray-700 py-1 px-3 pr-8 rounded leading-tight focus:outline-none focus:border-blue-500"
      >
        <option
          value="high-low"
        >
          Price high-low
        </option>
        <option
          value="low-high"
        >
          Price low-high
        </option>
      </select>
      <div
        class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
      >
        <svg
          class="fill-current h-4 w-4"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
